### PR TITLE
chore: Add cloud-node-manager v1.31.0

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

chore: Add cloud-node-manager v1.31.0

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/components.json [hunk 1]
```diff
@@ -224,7 +224,8 @@
         "v1.29.4",
         "v1.29.9",
         "v1.30.0",
-        "v1.30.5"
+        "v1.30.5",
+        "v1.31.0"
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/components.json:228]: can we start thinking about removing older versions?

VHD space is filling up quickly

Comment (nilo19) [reply parts/linux/cloud-init/artifacts/components.json:228]: Will remove once 1.27 is out

**Comment by nilo19** [comment]
Test failed, but not related to the change.
https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=102062723&view=logs&j=ceeb8f07-abdb-53f4-60d8-80121461daa9&t=4a7ed435-2f7e-5844-0d5d-22ae8168ab73&l=165

**Comment by nilo19** [comment]
@feiskyer could you help merge this?

**Review by feiskyer — APPROVED** [review]

**Review by ShiqianTao — APPROVED** [review]

**Pull request merged**